### PR TITLE
✨ transaction execBuy

### LINKED ISSUE #4942: composable transaction `transactionBuy`
### spec
- Create transactionBuy in composable/transaction

In GalleryItemBuy.vue

### PR DESCRIPTION
**Pull request by roiLeo**
This feature will handle `BUY` tx from one single point

## PR Type

- [x] Feature
- [x] Refactoring

## Context

- [x] Closes #4942 

#### Optional

- [x] I've tested it at `/snek/gallery/659233203-5`

#### side note

- `LIST` 0.0010
- `BUY` 0.0009
- ???

### CODE REVIEW

**Review by daiagi — COMMENTED** [review]
small stuff
overall code lgtm

**Review by daiagi — APPROVED** [review]

File: components/gallery/GalleryItemAction/GalleryItemActionType/GalleryItemBuy.vue [hunk 1]
```diff
@@ -207,12 +176,23 @@ const handleBuy = async () => {
...
-    showNotification(blockNumber, notificationTypes.info)
+  try {
+    transaction({
+      interaction: ShoppingActions.BUY,
+      currentOwner: props.currentOwner,
+      price: props.nftPrice,
+      nftId: $route.params.id,
+      tokenId: $route.params.id,
+      urlPrefix: urlPrefix.value,
+      successMessage: $i18n.t('mint.successNewNfts') as string,
```
[the remark this answers is not in the document]
Comment (roiLeo) [reply components/gallery/GalleryItemAction/GalleryItemActionType/GalleryItemBuy.vue]: works like a charm

**Comment by vikiival** [comment]
@roiLeo what does the side note mean?

**Comment by roiLeo** [comment]
> @roiLeo what does the side note mean?

On my test `/snek/gallery/659233203-5`
Capture d’écran 2023-03-03 à 11 51 55 AM

This is weird, there is a difference between `LIST` price and `BUY` price

**Comment by vikiival** [comment]
Tested on 

https://deploy-preview-5150--koda-nuxt.netlify.app/rmrk/gallery/9290733-94ae5001bac3833457-PNPRT-NFT_PINUP_GIRL_006-0000000000000005

WFM

I can't sadly buy any nfts on snek, some balance issue

**Comment by vikiival** [comment]
@roiLeo because royalty was calculated 

^-^

**Review by vikiival — APPROVED** [review]

**Comment by vikiival** [comment]
pay 44 usd

**Pull request merged**